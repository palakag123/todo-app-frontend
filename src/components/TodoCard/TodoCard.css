.todo-card{
    background-color: white;
    max-width: 95%;
    max-height: 110px;
    margin: 2.5%;
    border-radius: 10px;
    padding: 15px;
    padding-left: 30px;
}
#todo-title{
    font-size: 25px;
    color: rgb(87,204,237);
    font-weight: 600;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
}
@media only screen and (max-width: 770px) {
    #todo-title{
        font-size: 20px;
        color: rgb(87,204,237);
        font-weight: 600;
    }
}