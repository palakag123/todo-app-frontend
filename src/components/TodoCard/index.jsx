/* eslint-disable react/forbid-prop-types */
import React from "react";
import PropTypes from "prop-types";
import "./TodoCard.css";

function TodoCard({ todo }) {
  return (
    <div className="todo-card" data-testid="todoCard">
      <p id="todo-title">{todo?.todo}</p>
    </div>
  );
}

TodoCard.propTypes = {
  todo: PropTypes.object.isRequired,
};

export default TodoCard;
