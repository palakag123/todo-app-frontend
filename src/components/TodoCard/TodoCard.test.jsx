/* eslint-disable comma-dangle */
import React from "react";
import { render, screen } from "@testing-library/react";
import TodoCard from "./index";

describe("TodoCard", () => {
  const mockTodo = "Todo";
  const mockTodoCardId = 2;
  beforeEach(() => {
    render(
      <TodoCard
        todo={{ id: mockTodoCardId, todo: mockTodo }}
        key={mockTodoCardId}
      />
    );
  });
  it("should take a snapshot for UserCard", () => {
    const { asFragment } = render(
      <TodoCard
        todo={{ id: mockTodoCardId, todo: mockTodo }}
        key={mockTodoCardId}
      />
    );
    expect(
      asFragment(
        <TodoCard
          todo={{ id: mockTodoCardId, todo: mockTodo }}
          key={mockTodoCardId}
        />
      )
    ).toMatchSnapshot();
  });
  it("should have the required elements when UserCard is rendered", () => {
    expect(screen.getByTestId("todoCard")).toBeTruthy();
    expect(screen.getByText(mockTodo)).toBeTruthy();
  });
});
