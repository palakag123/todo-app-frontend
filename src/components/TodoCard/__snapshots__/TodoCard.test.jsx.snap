// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoCard should take a snapshot for UserCard 1`] = `
<DocumentFragment>
  <div
    class="todo-card"
    data-testid="todoCard"
  >
    <p
      id="todo-title"
    >
      Todo
    </p>
  </div>
</DocumentFragment>
`;
